File: SampleCode/RestApi/Clients/Java/src/main/java/EmbyClient/Java/UserViewsServiceApi.java
/*
 * Emby Server REST API
 * 
 */

package EmbyClient.Java;

import EmbyClient.ApiCallback;
import EmbyClient.ApiClient;
import EmbyClient.ApiException;
import EmbyClient.ApiResponse;
import EmbyClient.Configuration;
import EmbyClient.Pair;
import EmbyClient.ProgressRequestBody;
import EmbyClient.ProgressResponseBody;

import com.google.gson.reflect.TypeToken;

import java.io.IOException;


import io.swagger.client.model.QueryResultBaseItemDto;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class UserViewsServiceApi {
    private ApiClient apiClient;

    public UserViewsServiceApi() {
        this(Configuration.getDefaultApiClient());
    }

    public UserViewsServiceApi(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    public ApiClient getApiClient() {
        return apiClient;
    }

    public void setApiClient(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    /**
     * Build call for getUsersByUseridViews
     * @param userId User Id (required)
     * @param includeExternalContent Whether or not to include external views such as channels or live tv (required)
     * @param progressListener Progress listener
     * @param progressRequestListener Progress request listener
     * @return Call to execute
     * @throws ApiException If fail to serialize the request body object
     * API Documentation: Browsing the Library
     * @see <a href="https://dev.emby.media/doc/restapi/Browsing-the-Library.html"> Documentation</a>
     */
    public com.squareup.okhttp.Call getUsersByUseridViewsCall(String userId, Boolean includeExternalContent, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        Object localVarPostBody = null;
        
        // create path and map variables
        String localVarPath = "/Users/{UserId}/Views"
            .replaceAll("\\{" + "UserId" + "\\}", apiClient.escapeString(userId.toString()));

        List<Pair> localVarQueryParams = new ArrayList<Pair>();
        List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();
        if (includeExternalContent != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("IncludeExternalContent", includeExternalContent));

        Map<String, String> localVarHeaderParams = new HashMap<String, String>();

        Map<String, Object> localVarFormParams = new HashMap<String, Object>();

        final String[] localVarAccepts = {
            "application/json", "application/xml"
        };
        final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
        if (localVarAccept != null) localVarHeaderParams.put("Accept", localVarAccept);

        final String[] localVarContentTypes = {
            
        };
        final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
        localVarHeaderParams.put("Content-Type", localVarContentType);

        if(progressListener != null) {
            apiClient.getHttpClient().networkInterceptors().add(new com.squareup.okhttp.Interceptor() {
                @Override
                public com.squareup.okhttp.Response intercept(com.squareup.okhttp.Interceptor.Chain chain) throws IOException {
                    com.squareup.okhttp.Response originalResponse = chain.proceed(chain.request());
                    return originalResponse.newBuilder()
                    .body(new ProgressResponseBody(originalResponse.body(), progressListener))
                    .build();
                }
            });
        }

        String[] localVarAuthNames = new String[] { "apikeyauth", "embyauth" };
        return apiClient.buildCall(localVarPath, "GET", localVarQueryParams, localVarCollectionQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAuthNames, progressRequestListener);
    }
    
    @SuppressWarnings("rawtypes")
    private com.squareup.okhttp.Call getUsersByUseridViewsValidateBeforeCall(String userId, Boolean includeExternalContent, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        // verify the required parameter 'userId' is set
        if (userId == null) {
            throw new ApiException("Missing the required parameter 'userId' when calling getUsersByUseridViews(Async)");
        }
        // verify the required parameter 'includeExternalContent' is set
        if (includeExternalContent == null) {
            throw new ApiException("Missing the required parameter 'includeExternalContent' when calling getUsersByUseridViews(Async)");
        }
        
        com.squareup.okhttp.Call call = getUsersByUseridViewsCall(userId, includeExternalContent, progressListener, progressRequestListener);
        return call;

        
        
        
        
    }

    /**
     * 
     * Requires authentication as user
     * @param userId User Id (required)
     * @param includeExternalContent Whether or not to include external views such as channels or live tv (required)
     * @return QueryResultBaseItemDto
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     * API Documentation: Browsing the Library
     * @see <a href="https://dev.emby.media/doc/restapi/Browsing-the-Library.html"> Documentation</a>
     */
    public QueryResultBaseItemDto getUsersByUseridViews(String userId, Boolean includeExternalContent) throws ApiException {
        ApiResponse<QueryResultBaseItemDto> resp = getUsersByUseridViewsWithHttpInfo(userId, includeExternalContent);
        return resp.getData();
    }

    /**
     * 
     * Requires authentication as user
     * @param userId User Id (required)
     * @param includeExternalContent Whether or not to include external views such as channels or live tv (required)
     * @return ApiResponse&lt;QueryResultBaseItemDto&gt;
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     * API Documentation: Browsing the Library
     * @see <a href="https://dev.emby.media/doc/restapi/Browsing-the-Library.html"> Documentation</a>
     */
    public ApiResponse<QueryResultBaseItemDto> getUsersByUseridViewsWithHttpInfo(String userId, Boolean includeExternalContent) throws ApiException {
        com.squareup.okhttp.Call call = getUsersByUseridViewsValidateBeforeCall(userId, includeExternalContent, null, null);
        Type localVarReturnType = new TypeToken<QueryResultBaseItemDto>(){}.getType();
        return apiClient.execute(call, localVarReturnType);
    }

    /**
     *  (asynchronously)
     * Requires authentication as user
     * @param userId User Id (required)
     * @param includeExternalContent Whether or not to include external views such as channels or live tv (required)
     * @param callback The callback to be executed when the API call finishes
     * @return The request call
     * @throws ApiException If fail to process the API call, e.g. serializing the request body object
     * API Documentation: Browsing the Library
     * @see <a href="https://dev.emby.media/doc/restapi/Browsing-the-Library.html"> Documentation</a>
     */
    public com.squareup.okhttp.Call getUsersByUseridViewsAsync(String userId, Boolean includeExternalContent, final ApiCallback<QueryResultBaseItemDto> callback) throws ApiException {

        ProgressResponseBody.ProgressListener progressListener = null;
        ProgressRequestBody.ProgressRequestListener progressRequestListener = null;

        if (callback != null) {
            progressListener = new ProgressResponseBody.ProgressListener() {
                @Override
                public void update(long bytesRead, long contentLength, boolean done) {
                    callback.onDownloadProgress(bytesRead, contentLength, done);
                }
            };

            progressRequestListener = new ProgressRequestBody.ProgressRequestListener() {
                @Override
                public void onRequestProgress(long bytesWritten, long contentLength, boolean done) {
                    callback.onUploadProgress(bytesWritten, contentLength, done);
                }
            };
        }

        com.squareup.okhttp.Call call = getUsersByUseridViewsValidateBeforeCall(userId, includeExternalContent, progressListener, progressRequestListener);
        Type localVarReturnType = new TypeToken<QueryResultBaseItemDto>(){}.getType();
        apiClient.executeAsync(call, localVarReturnType, callback);
        return call;
    }
}

File: SampleCode/RestApi/Clients/Python/emby_client/api/user_views_service_api.py
# coding: utf-8

"""
    Emby Server REST API (BETA)
"""

from __future__ import absolute_import

import re  # noqa: F401

# python 2 and python 3 compatibility library
import six

from emby_client.api_client import ApiClient


class UserViewsServiceApi(object):
    """NOTE: This class is auto generated.
    """

    def __init__(self, api_client=None):
        if api_client is None:
            api_client = ApiClient()
        self.api_client = api_client

    def get_users_by_userid_views(self, user_id, include_external_content, **kwargs):  # noqa: E501
        """get_users_by_userid_views  # noqa: E501

        Requires authentication as user  # noqa: E501
        This method makes a synchronous HTTP request by default. To make an
        asynchronous HTTP request, please pass async_req=True
        >>> thread = api.get_users_by_userid_views(user_id, include_external_content, async_req=True)
        >>> result = thread.get()

        :param async_req bool
        :param str user_id: User Id (required)
        :param bool include_external_content: Whether or not to include external views such as channels or live tv (required)
        :return: QueryResultBaseItemDto
                 If the method is called asynchronously,
                 returns the request thread.
        """
        kwargs['_return_http_data_only'] = True
        if kwargs.get('async_req'):
            return self.get_users_by_userid_views_with_http_info(user_id, include_external_content, **kwargs)  # noqa: E501
        else:
            (data) = self.get_users_by_userid_views_with_http_info(user_id, include_external_content, **kwargs)  # noqa: E501
            return data

    def get_users_by_userid_views_with_http_info(self, user_id, include_external_content, **kwargs):  # noqa: E501
        """get_users_by_userid_views  # noqa: E501

        Requires authentication as user  # noqa: E501
        This method makes a synchronous HTTP request by default. To make an
        asynchronous HTTP request, please pass async_req=True
        >>> thread = api.get_users_by_userid_views_with_http_info(user_id, include_external_content, async_req=True)
        >>> result = thread.get()

        :param async_req bool
        :param str user_id: User Id (required)
        :param bool include_external_content: Whether or not to include external views such as channels or live tv (required)
        :return: QueryResultBaseItemDto
                 If the method is called asynchronously,
                 returns the request thread.
        """

        all_params = ['user_id', 'include_external_content']  # noqa: E501
        all_params.append('async_req')
        all_params.append('_return_http_data_only')
        all_params.append('_preload_content')
        all_params.append('_request_timeout')

        params = locals()
        for key, val in six.iteritems(params['kwargs']):
            if key not in all_params:
                raise TypeError(
                    "Got an unexpected keyword argument '%s'"
                    " to method get_users_by_userid_views" % key
                )
            params[key] = val
        del params['kwargs']
        # verify the required parameter 'user_id' is set
        if ('user_id' not in params or
                params['user_id'] is None):
            raise ValueError("Missing the required parameter `user_id` when calling `get_users_by_userid_views`")  # noqa: E501
        # verify the required parameter 'include_external_content' is set
        if ('include_external_content' not in params or
                params['include_external_content'] is None):
            raise ValueError("Missing the required parameter `include_external_content` when calling `get_users_by_userid_views`")  # noqa: E501

        collection_formats = {}

        path_params = {}
        if 'user_id' in params:
            path_params['UserId'] = params['user_id']  # noqa: E501

        query_params = []
        if 'include_external_content' in params:
            query_params.append(('IncludeExternalContent', params['include_external_content']))  # noqa: E501

        header_params = {}

        form_params = []
        local_var_files = {}

        body_params = None
        # HTTP header `Accept`
        header_params['Accept'] = self.api_client.select_header_accept(
            ['application/json', 'application/xml'])  # noqa: E501

        # Authentication setting
        auth_settings = ['apikeyauth', 'embyauth']  # noqa: E501

        return self.api_client.call_api(
            '/Users/{UserId}/Views', 'GET',
            path_params,
            query_params,
            header_params,
            body=body_params,
            post_params=form_params,
            files=local_var_files,
            response_type='QueryResultBaseItemDto',  # noqa: E501
            auth_settings=auth_settings,
            async_req=params.get('async_req'),
            _return_http_data_only=params.get('_return_http_data_only'),
            _preload_content=params.get('_preload_content', True),
            _request_timeout=params.get('_request_timeout'),
            collection_formats=collection_formats)
